### ISSUE TITLE
Test test_base.SpacepyDirTests fails on mac

### ISSUE THREAD

**Issue by spacecataz** [spacepy/spacepy#567]
My issue is: test_base.SpacepyDirTests.testDotflnRelative fails on Mac with Spacepy 0.2.3rc3.

### Minimal example to reproduce issue:

```
# Be on a mac.
python test_base.py
```

### Error message/Traceback:

```
======================================================================
FAIL: testDotflnRelative (test_base.SpacepyDirTests)
Checks DOT_FLN with a relative path
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/wellingdt/Downloads/spacepy-0.2.3rc3/tests/test_base.py", line 145, in testDotflnRelative
    self.assertEqual(os.path.join(self.td, '.spacepy'),
AssertionError: '/var/folders/lh/nh87pm7j0y3b7ng8scw8mww8f2p10y/T/tmp00ku2peh/.spacepy' != '/private/var/folders/lh/nh87pm7j0y3b7ng8scw8mww8f2p10y/T/tmp00ku2peh/.spacepy'
- /var/folders/lh/nh87pm7j0y3b7ng8scw8mww8f2p10y/T/tmp00ku2peh/.spacepy
+ /private/var/folders/lh/nh87pm7j0y3b7ng8scw8mww8f2p10y/T/tmp00ku2peh/.spacepy
? ++++++++
```

I was asked to run the following lines of code:
```
import tempfile
td = tempfile.mkdtemp()
print(td)
import os.path
print(os.path.abspath(td))
os.chdir(td)
print(os.path.abspath('.'))
```

It produced the following output:
```
/var/folders/lh/nh87pm7j0y3b7ng8scw8mww8f2p10y/T/tmpbpv6c69d
/var/folders/lh/nh87pm7j0y3b7ng8scw8mww8f2p10y/T/tmpbpv6c69d
/private/var/folders/lh/nh87pm7j0y3b7ng8scw8mww8f2p10y/T/tmpbpv6c69d
```

### OS, Python version, and dependency version information:

```
macOS-10.15.7-x86_64-i386-64bit
sys.version_info(major=3, minor=9, micro=7, releaselevel='final', serial=0)
numpy=1.21.3
scipy=1.7.1
matplotlib=3.4.3


```
-->

### Version of SpacePy
Spacepy 0.2.3rc3.

**Comment by jtniehof** [comment]
The functionality is fine here, it's the test. There's something weird where there's a symlink or bind mount or something and changing directory goes not to the directory named, but something that's followed the symlink or bind.

**Comment by jtniehof** [comment]
This was fixed in #639. I had forgotten we even had an open issue on it.
